docs/proposals/env-vars.md

```diff
@@ -0,0 +1,563 @@
+spec:
+  buildSteps: #Content omitted for this example
+  params:
+  - name: github-token-secret
+    description: Name of the secret holding the github-token.
+    default: github-token
+  environmentVariables:
+  - name: GITHUB_TOKEN
+    valueFrom:
+      secretKeyRef:
+        name: $(params.github-token-secret)
+        key: bot-token
+```
+
+Similar use of `params` and `environmentVariables` will exist in `Build` and `BuildRun` as well.
+
+The resulting `TaskRun`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: ...
+spec:
+  taskSpec:
+    steps:
+      - name: git-checkout
+        image: <you image of choice that containers 'git'>
+        env:
+          - name: GITHUB_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: <either a user supplied param value or the default 'github-token'>
+                key: bot-token
+
+```
+
+### User Stories [optional]
+
+#### Story 1
+
+As an image developer using Shipwright, I need to be able to control environment variables passed to the underlying
+tools used for image management, source code management, or data transfer, when I build image using Shipwright, assuming
+agreement with the cluster administrator has been obtained on which environment variables I can control.
+
+#### Story 2
+
+As a cluster administrator, I need to make sure that the users of my cluster do not use any unsafe environment variable enabled 
+features that exist with the underlying tools used for image management, source code management, or data transfer, when
+Shipwright runs in the cluster.
+
+#### Story 3
+
+As a cluster administrator, I need to make sure that only approved values for certain environment variables, which in turn control
+features that exist with the underlying tools used for image management, source code management, or data transfer, are used when
+Shipwright runs in the cluster.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+### Risks and Mitigations
+
+Coordination with the parameterization enhancement work is an added bit of complexity.
+
+## Design Details
+
+### Test Plan
+
+This requires new unit and integration tests.  In conjunction, documentation needs to be updated with explanations
+and examples.
+
+### Graduation Criteria
+
+Should be part of any release before our v1.0.0
+
+### Upgrade / Downgrade Strategy
+
+Nothing around environment variables exists yet in the API.  For changes after the initial implemenetation of 
+this proposal, see the references to the kubernetes recommendations elsewhere in this proposal.
+
+### Version Skew Strategy
+
+If this feature does not make v1alpha1, or if we change it in later versions, then the kubernetes recommendations around the use of annotations
+to deal with round trip transfer of `BuildStrategy`, `Build`, and `BuildRun` referenced in the [open questions](#open-questions-optional)
+section will be needed.
+
+## Implementation History
+
+
+## Drawbacks
+
+
```

Comment:
What are the drawbacks to this approach? Does our API and planned implementation preclude some future behavior?

Reply:
part of next push